### ISSUE TITLE
Too many dependencies are present after upgrading to Tycho 4.0.5.

### ISSUE THREAD

**Issue by reijnenf** [eclipse/dash-licenses#318]
After upgrading to Tycho 4.0.5 from 4.0.1 too many dependencies are returned, which are not in our product. See the newly added dependencies here: https://gitlab.eclipse.org/eclipse/escet/escet/-/merge_requests/788/diffs#cf3483dc93ae7e3c7a54201d846111014cbf504d and our discussion here: https://gitlab.eclipse.org/eclipse/escet/escet/-/merge_requests/788#note_1705986.

**Comment by waynebeaton** [comment]
The Maven plugin uses the standard Maven resolution mechanism. We're entirely at the mercy of how Eclipse Tycho interacts with the Maven reactor.

Are you invoking the tool with or without the `-Dtycho.target.eager=true` option?

**Comment by waynebeaton** [comment]
I get a much longer list of dependencies when I turn the `tycho.target.eager` on.

```
$ mvn dependency:tree -Dtycho.target.eager=false | grep -Poh "(?<=\+\-)[^:]+:[^:]+:[^:\s]+" | grep -v org\.eclipse | sort | uniq | wc -l
24
$ mvn dependency:tree -Dtycho.target.eager=true | grep -Poh "(?<=\+\-)[^:]+:[^:]+:[^:\s]+" | grep -v org\.eclipse | sort | uniq | wc -l
153
```

In my experience with Tycho-based builds, the `tycho.target.eager=true` option produces the more correct results. I don't recall offhand whether or not I've noticed source bundles (per #317) in the results in other cases.

**Comment by reijnenf** [comment]
Yes, we do use the `-Dtycho.target.eager=true` option, after #259.  Without, we get (after filtering) 28 dependencies, and with `eager=true` we get 742.
